source/robot_model/src/Model.cpp

```diff
@@ -346,4 +346,43 @@ void Model::print_qp_problem() {
     std::cout << this->upper_bound_constraints_(i) << std::endl;
   }
 }
+
+bool Model::in_range(const state_representation::JointPositions& joint_positions) {
```

Comment:
This is actually not needed as it is already covered in the `JointState` function. Same for the velocity and torque ones

Reply:
If we do like this, when we enter for example an object of type JointVelocities, the positions and the torques are set to zero, I guess. For the torque there are no problems, but for the joint positions there are, because it can have the limits that does not include the zero (e.g. `lower=-2` and `upper=-1`) In this case the function will return a `false` for a condition that depends on the positions and not on the velocities.

Reply:
Yes I see that but I haven't seen any model where that happens yet. The reason is the 0 is a design choice right? If a constructor wanted to actually do something like this one could simply shift the angles to have the 0 in range and the range as `lower=-0.5` and `upper=0.5`. You could even check that and have this offset stored. But I don't believe any model will show something like that.

Reply:
actually the 4th joint of franka emika is really close to have the 0 out the boundaries, but it is still inside:

```
... limit_lower="-3.0718" limit_upper="0.0698" ...
```
I have tried to use the function with JointState for all the other classes and it works properly, in the end which design you propose to use?  the one with only the function with JointState as input or the one with the different functions?

Reply:
actually in the URDF we are using for testing in the control libraries the 4th joint of Franka Emika is bounded as follow:
```
<limit effort="87" lower="-3.0718" upper="-0.0698" velocity="2.1750"/>
```
The zero is not included in the limits.

Reply:
It is weird that you don't have the same model already. If it can be an issue then we don't have any option than to duplicate code but I don't like it that much... In the state one I would actually propose to use casting for the other variables then:

`bool in_range(JointState joint_state) {
     return (in_range(static_cast<JointPositions&>(joint_state)) 
                && in_range(static_cast<JointVelocites&>(joint_state)
                && in_range(static_cast<JointTorques&>(joint_state));
}

`static_cast` cost 0 overhead and it removes a bit of code duplication.